lib/authorization.rb

```diff
@@ -22,11 +22,11 @@ def self.all_permissions(subject_id, object_id)
   end
 
   def self.authorized_resources_dataset(subject_id, actions)
-    matched_policies_dataset(subject_id, actions).select(:tagged_id)
+    matched_policies_dataset(subject_id, actions).select(Sequel[:applied_tag][:tagged_id])
   end
 
   def self.authorized_resources(subject_id, actions)
-    matched_policies_dataset(subject_id, actions).select_map(:tagged_id)
+    matched_policies_dataset(subject_id, actions).select_map(Sequel[:applied_tag][:tagged_id])
```

Comment:
Can `authorized_resources` use `authorized_resources_datase` like `authorized_resources_dataset(subject_id, actions).all`?

Reply:
You want it to return an array of values, not a hash.  You could use `authorized_resources_dataset(subject_id, actions).map(:tagged_id)`, but this approach is more efficient as it avoids a dataset allocation, and `select_map` is better optimized than `map` in sequel_pg.